aspnetcore/blazor/performance/profiling.md

```diff
@@ -0,0 +1,226 @@
+---
+title: ASP.NET Core Blazor WebAssembly performance profiling and diagnostic counters
+author: guardrex
+description: Learn about performance profiling and diagnostic counters in ASP.NET Core Blazor WebAssembly apps.
+monikerRange: '>= aspnetcore-10.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/16/2025
+uid: blazor/performance/profiling
+---
+# ASP.NET Core Blazor WebAssembly performance profiling and diagnostic counters
+
+<!-- UPDATE 10.0 - Activate ...
+
+[!INCLUDE[](~/includes/not-latest-version.md)]
+
+-->
+
+This article describes performance profiling tools and diagnostic counters for Blazor WebAssembly apps.
+
+## Prerequisite
+
+Install the [.NET WebAssembly build tools](xref:blazor/tooling/webassembly#net-webassembly-build-tools):
+
+```dotnetcli
+dotnet workload install wasm-tools
+```
+
+## Browser developer tools
+
+App code can be manually profiled using the performance profiler in a browser's developer tools console.
+
+Built-in performance counters are available to track:
+
+* [Ahead-of-time (AOT) compilation](xref:blazor/tooling/webassembly#ahead-of-time-aot-compilation)
+* Code interpolation
+* [JIT (Just-In-Time) interpolation](https://developer.mozilla.org/docs/Glossary/Just_In_Time_Compilation) 
+* Call specification (":::no-loc text="callspec":::", sequence and timing of function calls) and instrumentation
+
+Enable integration with the browser's developer tools profiler using the `<WasmProfilers>` property in the app's project file (`.csproj`). Include the additional properties in the following table.
+
+Property | Default | Set value to&hellip; | Description
+--- | :---: | :---: | ---
+`<WasmProfilers>` | No value | `browser` | Mono profilers to use. Potential values are "`browser`" and "`log`". To use both, separate the values with a semicolon. The `browser` profiler enables integration with the browser's developer tools profiler.
+`<RunAOTCompilation>`| `false` | `true` | Controls AOT compilation.
+`<RunAOTCompilationAfterBuild>` | `false` | `true` | Controls AOT compilation after build. By default, it's run only for publish.
+`<WasmNativeStrip>` | `true` | `false` | Controls stripping the native executable.
+`<WasmNativeDebugSymbols>` | `true` | `true` | Controls building with native debug symbols.
+`<WasmBuildNative>` | `false` | `true` | Controls building the native executable.
+
+```xml
+<PropertyGroup>
+  <WasmProfilers>browser;</WasmProfilers>
```

Comment:
You can use `<WasmProfilers>browser;</WasmProfilers>` without AOT. 

Also you can dev-tools profiler with AOT and without `<WasmProfilers>browser;</WasmProfilers>`.
To be able to see AOT method names in the browser tools, you need to install [DWARF chrome extension](https://chromewebstore.google.com/detail/cc++-devtools-support-dwa/pdcpmagijalfljmkmjngeonclgbbannb)